# Exploratory Data Analysis
Get a general understanding of the dataset provided and what potential issues to handle.

In [1]:
import pandas as pd
import numpy as np
from pathlib import Path

In [2]:
data_dir = Path.cwd().parent / 'LUNA'

In [3]:
candidates_df = pd.read_csv(data_dir / 'candidates.csv')

In [4]:
candidates_df.head()

seriesuid  coordX  coordY  coordZ  \
0  1.3.6.1.4.1.14519.5.2.1.6279.6001.100225287222...  -56.08  -67.85 -311.92   
1  1.3.6.1.4.1.14519.5.2.1.6279.6001.100225287222...   53.21 -244.41 -245.17   
2  1.3.6.1.4.1.14519.5.2.1.6279.6001.100225287222...  103.66 -121.80 -286.62   
3  1.3.6.1.4.1.14519.5.2.1.6279.6001.100225287222...  -33.66  -72.75 -308.41   
4  1.3.6.1.4.1.14519.5.2.1.6279.6001.100225287222...  -32.25  -85.36 -362.51   

   class  
0      0  
1      0  
2      0  
3      0  
4      0

In [5]:
candidates_df.shape

(551065, 5)

There are 551065 lumps, i.e. potential malignant lumps in the dataset.

---
Columns:
- `seriesuid` the UID in DICOM format,
- `XYZ` coordinates,
- `class`, this corresponds to the nodule status (boolean value: 0 not actual nodule, and 1 for a nodule, either malignant or benign)

In [8]:
candidates_df['class'].unique()

array([0, 1], dtype=int64)

In [9]:
# Counts for each class
candidates_df['class'].value_counts()

0    549714
1      1351
Name: class, dtype: int64

Of all the potential candidates, 1351 are malignant. These are then annotated in the `annotations.csv`.

In [13]:
candidates_df.seriesuid.nunique()

888